### PR TITLE
Use different IPFS ports for different channels

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/12369

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`, `npm run gn_check`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Requested a security/privacy review as needed.
- [x] Added appropriate QA labels (`QA/Yes` or `QA/No`) to the associated issue
- [x] Added appropriate release note labels (`release-notes/include` or `release-notes/exclude`) to the associated issue
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Review by bbondy — APPROVED** [review]
Looks good!

**Comment by yrliou** [comment]
Windows failed at unrelated rewards browsertests.

File: components/ipfs/translate_ipfs_uri.cc [hunk 1]
```diff
@@ -42,9 +42,7 @@ bool TranslateIPFSURI(const GURL& url, GURL* new_url, bool local) {
       // new_url would be:
       // https://dweb.link/ipfs/[cid]//wiki/Vincent_van_Gogh.html
       if (new_url) {
-        std::string gateway_empty_path =
-            local ? GetDefaultIPFSLocalGateway().spec()
-                  : GetDefaultIPFSGateway().spec();
+        std::string gateway_empty_path = gateway_url.spec();
         if (gateway_empty_path.size() > 0 &&
             gateway_empty_path[gateway_empty_path.size() - 1] == '/') {
```
Comment (iefremov) [inline components/ipfs/translate_ipfs_uri.cc:47]: not part of this PR, but couldn't we use `GURL::Resolve()` for these manipulations?

File: components/ipfs/ipfs_service.h [hunk 2]
```diff
@@ -49,7 +50,8 @@ class IpfsService : public KeyedService,
  public:
   explicit IpfsService(content::BrowserContext* context,
```
Comment (iefremov) [inline components/ipfs/ipfs_service.h:51]: isn't `explicit` anymore

File: components/ipfs/ipfs_ports.cc [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/ipfs/ipfs_ports.h"
+
+#include "base/strings/string_number_conversions.h"
+
+namespace {
+
+// Default ports used for non-official builds.
```
Comment (iefremov) [inline components/ipfs/ipfs_ports.cc:12]: hmm, where do we check for official vs non-official?.. Aren't these used for any builds?

**Review by iefremov — APPROVED** [review]

File: components/ipfs/ipfs_ports.cc (hunk 3 above)
Comment (yrliou) [reply components/ipfs/ipfs_ports.cc]: Comment updated.

File: components/ipfs/ipfs_service.h (hunk 2 above)
Comment (yrliou) [reply components/ipfs/ipfs_service.h]: Removed

**Comment by yrliou** [comment]
Can't get windows CI to run but it was already green before trivial changes for addressing review comments, merging.
